src/Component/Metadata/AsFilter.php

```diff
@@ -0,0 +1,28 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Component\Grid\Metadata;
+
+#[\Attribute(\Attribute::TARGET_CLASS)]
+final class AsFilter
+{
+    /**
+     * @param string $formType The form type class name to use for filter rendering
+     * @param string|null $type The filter type (defaults to FQCN of the class)
+     */
+    public function __construct(
+        public string $formType,
+        public ?string $type = null,
```

Comment:
```suggestion
        private string $formType,
        private ?string $type = null,
```

Reply:
```php
    public function getFormType(): string
    {
        return $this->formType;
    }

    public function getType(): ?string
    {
        return $this->type;
    }
```

Reply:
public is perfect for attributes
https://github.com/symfony/symfony/blob/7.2/src/Symfony/Component/Console/Attribute/AsCommand.php

Reply:
I also like the public approach more but we can't use readonly on them because we set these field while processing them

Reply:
hum... ok, but public is ok for me.

Reply:
So I changed to public not readonly as both loic and Łukasz agreed